src/lib/components/EventMap/index.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @fileoverview Element that renders a map and a list of local events.
+ */
+
+import {html} from 'lit-element';
+import {BaseElement} from '../BaseElement';
+import './_styles.scss';
+
+const encodeMarkers = function (markers) {
+  const color = 'red';
+  return markers
+    .map((marker) => {
+      return `markers=color:${color}%7Clabel:${marker.place[0]}%7C${marker.marker}`;
+    })
+    .join('&');
+};
+
+const formatDate = function (date) {
+  const options = {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+  };
+  return new Intl.DateTimeFormat('en-US', options).format(date);
+};
+
+/**
+ * Renders a map and a list of local events. It takes input from a <code>
+ * element embedded as a child of this element.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class EventMap extends BaseElement {
+  static get properties() {
+    return {
+      title: {type: String},
+      center: {type: String},
+      zoom: {type: Number},
+      size: {type: String},
+      events: {type: String},
+      key: {type: String},
+      localEvents: {type: Object},
+    };
+  }
+
+  constructor() {
+    super();
+    this.title = '';
+    this.center = '';
+    this.zoom = 4;
+    this.size = '600x300';
+    this.events = [];
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.events = JSON.parse(this.querySelector('code').innerText.trim());
```

Comment:
Sam is working on an element which will take all of the data from event.js and merge it into the unistore store. Originally he was just doing `days` but [I added a comment](https://github.com/GoogleChrome/web.dev/pull/3268/files#r442457600) that your maps widget will need data as well.

I think what this element could do instead is extend BaseStateElement and override the onStateChanged method and use that to trigger its render. Something like:

```
onStateChanged({event}) {
  if (!event) {
    return;
  }

  this.events = event.communityEvents;
  // render happens
}
```